### PR TITLE
Draft implementation of topological ordering

### PR DESCRIPTION
**Pull request by garyanaplan**
I've tried to implement this in a style that is sympathetic with the existing style in the Dfs/Bfs implementations.

Very happy to make changes or accept suggestions for improvements.

### CODE REVIEW

File: src/graph.rs [hunk 1]
```diff
@@ -974,6 +974,43 @@ impl<T> Graph<T> {
         Bfs::new(self)
     }
 
+    /// Returns an iterator over the vertices
+    /// of the graph which follows a DFS based
+    /// topological order (Kahn's algorithm).
+    ///
+    /// Topological sorting is not possible for
```
Comment (octavonce) [inline src/graph.rs:981]:
```
Topological sorting is not possible for
graphs which contain a cycle. You may
use topo.is_cylic() == false to verify
that your graph is a DAG.
```

What happens if the user doesn't check this? Maybe we should add a panic?

File: src/iterators/topo.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+use alloc::vec::Vec;
+
+#[derive(Debug)]
+/// Topological Iterator
+pub struct Topo<'a, T> {
+    /// The Graph being iterated.
+    iterable: &'a Graph<T>,
+    /// Processed vertices
+    vertices: Vec<&'a VertexId>,
+    /// Working set of vertoces
```
Comment (octavonce) [inline src/iterators/topo.rs]: Doc typo

**Review by octavonce — COMMENTED** [review]
Just two minor things but besides those, it looks good!

File: src/graph.rs (hunk 1 above)
Comment (garyanaplan) [reply src/graph.rs:981]: You are right, that's a problem.

I've pushed a fix and added a test to exercise the panic code. I can't figure out how to make the test check for PANIC_MSG, so I hard-coded the text into the test. That's a bit annoying.

**Comment by garyanaplan** [comment]
Ok I think the documentation change and the addition of the panic!() is all I want to do on this right now.

**Review by octavonce — APPROVED** [review]
Ok, this looks great! Merging now.

**Pull request merged**